pandapipes/std_types/std_type_class.py

```diff
@@ -35,120 +39,212 @@ def from_dict(cls, d):
-        :param path: Path where the CSV file, defining a pump object, is stored
-        :type path: str
-        :return: An object of the pump standard type class
-        :rtype: PumpStdType
-        """
-        p_values, v_values, degree = get_p_v_values(path)
-        reg_par = regression_function(p_values, v_values, degree)
-        pump_st = cls(name, reg_par)
-        pump_st._pressure_list = p_values
-        pump_st._flowrate_list = v_values
-        pump_st._reg_polynomial_degree = degree
-        return pump_st
+        reg_par, x_values, y_values, degree = cls._from_path(path)
+        reg_st = cls(name, reg_par)
+        cls.init_std_type(reg_st, x_values, y_values, degree)
+        return reg_st
 
     @classmethod
-    def from_list(cls, name, p_values, v_values, degree):
-        reg_par = regression_function(p_values, v_values, degree)
-        pump_st = cls(name, reg_par)
-        pump_st._pressure_list = p_values
-        pump_st._flowrate_list = v_values
-        pump_st._reg_polynomial_degree = degree
-        return pump_st
+    def from_list(cls, name, x_values, y_values, degree):
+        reg_par, x_values, y_values, degree = cls._from_list(x_values, y_values, degree)
+        reg_st = cls(name, reg_par)
+        cls.init_std_type(reg_st, x_values, y_values, degree)
 
+    @classmethod
+    def load_data(cls, path):
+        """
+        load_data.
 
-def get_data(path, std_type_category):
-    """
-    get_data.
-
-    :param path:
-    :type path:
-    :param std_type_category:
-    :type std_type_category:
-    :return:
-    :rtype:
-    """
-    if std_type_category == 'pump':
+        :param path:
+        :type path:
+        :return:
+        :rtype:
+        """
         path = os.path.join(path)
         data = pd.read_csv(path, sep=';', dtype=np.float64)
-    elif std_type_category == 'pipe':
-        data = pd.read_csv(path, sep=';', index_col=0).T
-    else:
-        raise AttributeError('std_type_category %s not implemented yet' % std_type_category)
-    return data
+        return data
 
 
-def get_p_v_values(path):
+def regression_function(x_values, y_values, degree):
     """
+    Regression function...
 
-    :param path:
-    :type path:
+    :param p_values:
+    :type p_values:
+    :param v_values:
+    :type v_values:
+    :param degree:
+    :type degree:
     :return:
     :rtype:
     """
-    data = get_data(path, 'pump')
-    p_values = data.values[:, 0]
-    v_values = data.values[:, 1]
-    degree = data.values[0, 2]
-    return p_values, v_values, degree
+    if not int(degree) == degree:
+        raise UserWarning("The polynomial degree has to be an integer, but %s was given. "
+                          "It will be rounded down now." % str(degree))
+    return np.polyfit(x_values, y_values, degree)
 
 
-def regression_function(p_values, v_values, degree):
+def interpolation_function(x_values, y_values, fill_value='extrapolate'):
     """
     Regression function...
```

Comment:
docstring is outdated. Wrong parameter names and "regression function" should be "interpolation... "

Reply:
updated